src/ThreeEditor/js/Viewport.ClipPlane.js

```diff
@@ -0,0 +1,165 @@
+import * as THREE from 'three';
+import { GUI } from 'three/examples//jsm/libs/dat.gui.module.js';
+
+export function ViewportClippedView(editor, viewport, planeHelpers, initialObjects, signalGeometryChanged, { clipPlane, planeHelperColor, planePosLabel }) {
+
+    const STENCIL_RENDER_ORDER = 1;
+    const CLIPPING_SIZE = 10;
+
+    this.scene = new THREE.Scene();
+    this.gui = new GUI({});
+
+    // Setup plane clipping ui  
+    const planeHelper = new THREE.PlaneHelper(clipPlane, CLIPPING_SIZE, planeHelperColor ?? 0xffff00); // default helper color is yellow
+    planeHelpers.add(planeHelper);
+
+    const planePosProperty = planePosLabel ?? 'PlanePos';
+
+    const uiProps = {
+
+        get [planePosProperty]() {
+
+            return clipPlane.constant;
+
+        },
+        set [planePosProperty](v) {
+
+            crossSectionPlane.translateZ(v - clipPlane.constant); // translate Cross Section Plane with clip plane
```

Comment:
Explain why do we translate along Z axis, not along the X or Y ?
Also `translate ... with ...` doesn't make more sense (dramatically). We should rather `translate from ... to ...`  or `translate by ... in direction of ...`